app/code/Magento/Sitemap/etc/adminhtml/system.xml

```diff
@@ -83,10 +83,12 @@
                 <label>Sitemap File Limits</label>
                 <field id="max_lines" translate="label" type="text" sortOrder="1" showInDefault="1" showInWebsite="1" showInStore="1" canRestore="1">
                     <label>Maximum No of URLs Per File</label>
+                    <validate>validate-number</validate>
```

Comment:
There is another validation class "validate-greater-than-zero", that should fix the problem with values below or equal to zero.

Reply:
I have added validation for greater than zero